Extension does not detect the absence of ocamllsp

When using a switch that does not have ocaml-lsp-server installed, 
![image](https://user-images.githubusercontent.com/40719961/113432998-a9756d00-93de-11eb-878b-914834dcf547.png)

Under the "OCaml Language Server" output, we can see
```
[ERROR] Command not found 'ocamllsp'
[Info  - 18:09:36] Connection to server got closed. Server will restart.
[ERROR] Command not found 'ocamllsp'
[Info  - 18:09:36] Connection to server got closed. Server will restart.
[ERROR] Command not found 'ocamllsp'
[Info  - 18:09:36] Connection to server got closed. Server will restart.
[ERROR] Command not found 'ocamllsp'
[Info  - 18:09:36] Connection to server got closed. Server will restart.
[ERROR] Command not found 'ocamllsp'
[Error - 18:09:36] Connection to server got closed. Server will not be restarted.
```

Yet the command `which` is called, so you maybe forgot to act on the fact that it does not find ocamllsp

```
external command (finished) {
  "bin": "/usr/local/bin/opam",
  "args": [
    "exec",
    "--switch=test2",
    "--root",
    "/home/me/.opam",
    "--",
    "which",
    "ocamllsp"
  ],
  "result": {
    "exitCode": 1,
    "stdout": "",
    "stderr": "which: no ocamllsp in (/home/me/.opam/test2/bin:/home/me/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:[rest of $PATH])\n"
  }
}

external command {
  "bin": "/usr/local/bin/opam",
  "args": [
    "exec",
    "--switch=test2",
    "--root",
    "/home/me/.opam",
    "--",
    "ocamllsp"
  ]
}
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Change Log
 
+## Unreleased
+
+- Fix showing error message if ocaml-lsp is missing (#586)
+
 ## 1.8.2
 
 - Fix switching between implementation/interface (#561)

--- a/src/extension_instance.ml
+++ b/src/extension_instance.ml
@@ -51,10 +51,9 @@ let start_language_server t =
   stop_server t;
   let res =
     let open Promise.Result.Syntax in
-    let* _ =
-      let+ has_command =
-        Sandbox.has_command t.sandbox "ocamllsp" |> Promise.map Result.return
-      in
+    let* () =
+      let open Promise.Syntax in
+      let+ has_command = Sandbox.has_command t.sandbox "ocamllsp" in
       if has_command then
         Ok ()
       else

